crates/bevy_sprite/src/texture_atlas_builder.rs

```diff
@@ -86,8 +86,9 @@ impl<'a> TextureAtlasBuilder<'a> {
     ///
     /// Optionally an asset id can be passed that can later be used with the texture layout to retrieve the index of this texture.
     /// The insertion order will reflect the index of the added texture in the finished texture atlas.
-    pub fn add_texture(&mut self, image_id: Option<AssetId<Image>>, texture: &'a Image) {
+    pub fn add_texture(mut self, image_id: Option<AssetId<Image>>, texture: &'a Image) -> Self {
```

Comment:
To be idiomatic and not awkward in cases like [these](https://github.com/bevyengine/bevy/pull/13238/files#diff-d7502361a8d4ac2d76dfc9c104a8811ed68238789590dc683c28d1deed09648eL222), this should return `&mut Self`: 

```rust
pub fn add_texture(&mut self, image_id: Option<AssetId<Image>>, texture: &'a Image) -> &mut Self
```

Reply:
>  It is often more ergonomic (and more efficient) to take and return the builder as a mutable reference. The borrow checker makes this work naturally. This approach has the advantage that one can write code like
>
> ```
> let mut fb = FooBuilder::new();
> fb.a();
> fb.b();
> let f = fb.build();
> ```
> as well as the FooBuilder::new().a().b().build() style.

From https://rust-unofficial.github.io/patterns/patterns/creational/builder.html#discussion

Imo, the only way that the `builder = builder.method(...);` pattern makes sense is if the method were named something less imperative like `with_texture(...)`, but as @IceSentry mentioned, that style of use just makes things easier to mess up with no gain.

Is there a practical case this PR is attempting to address?